nsq/protocol.py

```diff
@@ -70,15 +71,24 @@ def decode_message(data):
 
 
 def _command(cmd, body, *params):
-    body_data = ''
-    params_data = ''
+    body_prefix = b''
+    params_data = b''
     if body:
-        assert isinstance(body, str), 'body must be a string'
-        body_data = struct.pack('>l', len(body)) + body
+        assert isinstance(body, compat.string_like), 'body must be a string'
+        if isinstance(body, compat.unicode):
+            body = body.encode('utf-8')
```

Comment:
We had a discussion about an approach like this a long time ago, I'm still of the opinion that we should not be encoding the body passed in here and instead should assert bytes (or `str`).